### PR TITLE
Optimizes Native FSDP memory_efficient_init weight loading for multi-node EP/FSDP jobs & add mutlti-node scripts

### PR DESCRIPTION
**Pull request by meichangsu1**
# PR type
- [x] Bug Fix
- [ ] New Feature
- [ ] Document Updates
- [ ] More Models or Datasets Support

# PR information

This PR improves EP + FSDP2 + LoRA training for DeepSeek V4 Flash in multi-node environments.

Main changes:

1. Optimize Native FSDP `memory_efficient_init` weight loading for multi-node EP/FSDP jobs.
   Previously, global rank 0 distributed pretrained weights and EP expert shards to all ranks. With large `world_size`, especially when `ep_fsdp_size=1`, this put heavy communication pressure on global rank 0.
   This PR changes the loading path so each node's local rank 0 loads/captures the full pretrained state and distributes tensors only to ranks on the same node. EP shard selection still uses `rank_to_ep_rank`, so each target rank receives the correct EP slice before FSDP/DTensor placement.

2. Optimize LoRA checkpoint saving.
   LoRA adapter saving no longer calls full-model `get_full_state_dict()` and then filters LoRA keys. Instead, it collects only LoRA adapter parameters. Native FSDP keeps EP-aware LoRA all-gather, avoiding large base-model state_dict materialization during adapter checkpoint saving.

3. Remove the Twinkle-side Accelerate FSDP2 state-dict loading monkey patch.
   `AccelerateStrategy` now relies on native Accelerate behavior for `memory_efficient_init` / `cpu_ram_efficient_loading`.

4. Add multi-node DeepSeek V4 Flash EP + FSDP2 + LoRA cookbook script.
   The DeepSeek V4 LoRA cookbook now supports configuring GPU/NPU count via `NUM_GPUS`.

## Experiment results
[2026-06-01 07:35:49][INFO:twinkle] Current is step 4 of 16, metric: {'loss': '3.0792', 'learning rate(param group 1)': '0.000000e+00', 'learning rate(param group 2)': '0.000000e+00', 'iters': 0, 'total time elapse': '2.9 minutes', 'speed': '0.00 iters/s'}
[2026-06-01 07:36:16][INFO:twinkle] Current is step 8 of 16, metric: {'loss': '2.9792', 'grad_norm': '135.397873', 'learning rate(param group 1)': '2.000000e-05', 'learning rate(param group 2)': '2.000000e-05', 'iters': 1, 'total time elapse': '200 seconds', 'speed': '0.04 iters/s'}
[2026-06-01 07:36:51][INFO:twinkle] Current is step 12 of 16, metric: {'loss': '3.0201', 'grad_norm': '136.174759', 'learning rate(param group 1)': '4.000000e-05', 'learning rate(param group 2)': '4.000000e-05', 'iters': 2, 'total time elapse': '236 seconds', 'speed': '0.03 iters/s'}
[2026-06-01 07:37:22][INFO:twinkle] Saved final adapter to /nas/diskz/checkpoint-final

### CODE REVIEW

File: src/twinkle/model/transformers/strategy/accelerate.py
Comment (tpx818) [inline src/twinkle/model/transformers/strategy/accelerate.py:1]: _patch_accelerate_fsdp2_load_full_state_dict，这段patch现在不需要了吗

Comment (meichangsu1) [reply src/twinkle/model/transformers/strategy/accelerate.py:1]: 是的，这段逻辑最好由accelerate本身去处理，之前这段逻辑主要是为了把流程跑通

**Review by tpx818 — APPROVED** [review]
